src/ModelManager.ts

```diff
@@ -130,91 +139,160 @@ export class ModelManager {
      * @return {Promise}
      */
     public initialize<M extends Model>(config?: ModelManagerConfiguration | string): Promise<M> {
-        this.destroy();
+        this.initializeAsync(config);
+        const { rootModelURL, rootModelPath } = this._modelPaths;
 
-        let path;
-        let initialModel = null;
+        if (!rootModelURL) {
+            throw new Error("ModelManager.js Cannot initialize without a URL to fetch the root model");
+        }
 
-        if (!config || (typeof config === 'string')) {
-            path = config;
-        } else if (config) {
-            path = config.path;
-            this._modelClient = config.modelClient;
-            initialModel = config.model;
+        if (!rootModelPath) {
+            throw new Error('No root modelpath resolved! This should never happen.');
         }
 
-        this._listenersMap = {};
-        this._fetchPromises = {};
-        this._initPromise = null;
+        return this._initPromise;
+    }
 
-        const pageModelRoot = PathUtils.getMetaPropertyValue(MetaProperty.PAGE_MODEL_ROOT_URL);
-        const metaPropertyModelUrl = PathUtils.internalize(pageModelRoot);
-        const currentPathname = PathUtils.getCurrentPathname();
-        const sanitizedCurrentPathname = currentPathname ? PathUtils.sanitize(currentPathname) : '';
+    /**
+     * Initializes the ModelManager asynchronously using the rootModelPath to resolve a data model.
+     * For remote apps with no defined root model, an empty store is initialized and data is fetched on demand by components.
+     *
+     * Once the initial model is loaded and if the data model doesn't contain the path of the current pathname,
+     * the library attempts to fetch a fragment of model.
+     *
+     * @param {string|InitializationConfig} [config]                - URL to the data model or configuration object
+     * @fires cq-pagemodel-loaded if root model path is available
+     */
+    public initializeAsync(config?: ModelManagerConfiguration | string): void {
```

Comment:
- initialize: Have mostly left it intact to not affect existing implementations, hence why created a new api initializeAsync
- Previously, render was done only when the returned promise was resolved.
- initializeAsync: Ideally this handles all the initializations ie. modelclient, modelstore etc independent of model fetching. 
- The promise is returned only when there is a root path available. This will not be true in the case of our remote app when we are fetching component models on demand. No root path available => no root model fetched => nothing to return. 
- Also, In the current scenario, we don't need to return anything anymore since the fetched model will already be stored, and returning the model/promise on initialization is now obsolete.